Libraries/LibJS/Runtime/ArrayBuffer.h

```diff
@@ -35,10 +35,24 @@ struct DataBlock {
         Yes,
     };
 
+    struct UnownedFixedLengthByteBuffer {
+        UnownedFixedLengthByteBuffer(ByteBuffer* buffer)
+            : buffer(buffer)
+            , size(buffer ? buffer->size() : 0)
+        {
+        }
+
+        ByteBuffer* buffer;
+        size_t size;
+    };
+
     ByteBuffer& buffer()
     {
         ByteBuffer* ptr { nullptr };
-        byte_buffer.visit([&](Empty) { VERIFY_NOT_REACHED(); }, [&](auto* pointer) { ptr = pointer; }, [&](auto& value) { ptr = &value; });
+        byte_buffer.visit(
+            [&](Empty) { VERIFY_NOT_REACHED(); },
+            [&](ByteBuffer& value) { ptr = &value; },
+            [&](UnownedFixedLengthByteBuffer& value) { ptr = value.buffer; });
         return *ptr;
```

Comment:
Now that we're here,
```suggestion
        return byte_buffer.visit(
            [](Empty) -> ByteBuffer* { VERIFY_NOT_REACHED(); },
            [](ByteBuffer& value) { return &value; },
            [](UnownedFixedLengthByteBuffer& value) { return value.buffer; });
```